validation/genesis-predeploy_test.go

```diff
@@ -0,0 +1,121 @@
+package validation
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/require"
+)
+
+// Define a struct to represent the structure of the JSON data
+type DeployedBytecode struct {
+	Object              string                          `json:"object"`
+	ImmutableReferences map[string][]ImmutableReference `json:"immutableReferences"`
+}
+
+type ImmutableReference struct {
+	Start  int `json:"start"`
+	Length int `json:"length"`
+}
+
+type ContractData struct {
+	DeployedBytecode DeployedBytecode `json:"deployedBytecode"`
+}
+
+// Invoke this with go test -timeout 0 ./validation -run=TestGenesisPredeploys -v
```

Comment:
This passes for me locally, it should be straightforward to trigger it in on CI (just need to clone the monorepo to a neighbouring directory).